src/core/libmaven/PeakDetector.cpp

```diff
@@ -253,24 +253,43 @@ vector<mzSlice*> PeakDetector::processCompounds(vector<Compound*> set,
         return slices;
 }
 
-void PeakDetector::alignSamples() {
-        //being called in CLI.
-        if (mavenParameters->samples.size() > 1
-            && mavenParameters->alignSamplesFlag) {
-                cerr << "Aligning samples" << endl;
-
-                mavenParameters->writeCSVFlag = false;
-                processMassSlices();
-
-                cerr << "Aligner=" << mavenParameters->allgroups.size() << endl;
-                vector<PeakGroup*> agroups(mavenParameters->allgroups.size());
-                for (unsigned int i = 0; i < mavenParameters->allgroups.size(); i++)
-                        agroups[i] = &mavenParameters->allgroups[i];
-                //init aligner
-                Aligner aligner;
-                aligner.doAlignment(agroups);
-                mavenParameters->writeCSVFlag = true;
+void PeakDetector::alignSamples(const int& method) {
+
+    // only called from CLI
+    if(mavenParameters->samples.size() > 1 ) {
+
+        switch(method) {
+
+        case 1: {
+            ObiParams params("cor", false, 2.0, 1.0, 0.20, 3.40, 0.0, 20.0, false, 0.60);
```

Comment:
@rish9511 where are these arguments coming from?

Reply:
I'm not too comfortable hard-coding these values here. Might update the defaults in the ui file because of some issue and forget to update them in this function. But I can't see a good way to solve this either. Your call.

Reply:
Just by looking at it, default constructor seems to be the best place to define such values
But again, i have not thought about this well enough